catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/ws/WebSocketAuthenticationException.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.ws;
+
+/** An exception to represent a security-related error in the {@link SecureWebSocketServlet}. */
+public class WebSocketAuthenticationException extends RuntimeException {
```

Comment:
Why do we need a custom exception type?

Reply:
Primarily to make sure the `JsonRpc` implementation can know the exception was specific to our WebSocket implementation about the user being logged out. https://github.com/codice/ddf/pull/4325/files/0979939c23d25296551b3f51c6e2feb331111306#diff-456bad44a258908644aaed4370a6c8f8R104